src/components/input/index.tsx

```diff
@@ -99,13 +104,39 @@ const Input = forwardRef<HTMLInputElement, Props>(
       visible: !!value,
       disabled,
       onClear: () => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ;(inputRef.current as any).value = ""
         onChange({ target: { name, value: "", cleared: true } })
       },
     }
 
+    const debouncedChangeHandler = useMemo(() => {
+      if (onChange) {
+        return debounce(onChange, delay)
+      }
+      return null
+    }, [])
+
+    useEffect(() => {
+      return () => {
+        debouncedChangeHandler?.cancel()
+      }
+    }, [])
+
+    useEffect(() => {
+      if (ref) {
+        if (typeof ref === "function") {
+          ref(inputRef.current)
+        } else {
```

Comment:
can you shed light on the first branch of the conditional? when would ref not be a function? can we guard those cases with types instead of conditionals in the code?

Reply:
because refs can be passed as functions and objects.
```
myRef = React.createRef();
<>
  <input ref={this.myRef} />
  // Or
  <input ref={(ref) => (this.myRef.current = ref)} />
</>;
```
So, depending on what it's type is, we use a different way to assign its value

Reply:
https://github.com/Microsoft/TypeScript/issues/2072
I'ts not possible to use `instanceof` for `Function` type, it looks like its by design. 
```
typeof ref === "function"
```
doesn't look to bad  IMO though

Reply:
isn't the `ref` completely internal to this component? we can pretty much guarantee it's always one of the types, no?

Reply:
We didn't remove the forwardRef from the component, meaning that the consumer apps can still pass in `refs`. In those cases we combine the passed one with the internal one

Reply:
did you find any usage of the forwarded ref in the projects? last time I did a quick check, we didn't have one. so I'd consider dropping it so we have to support less branches

Reply:
I didn't find either. Sure, we can drop it

Reply:
we discussed to keep it